cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaConfigurationDiff.java

```diff
@@ -65,27 +67,164 @@ public class KafkaBrokerConfigurationDiff extends AbstractJsonDiff {
+            "sasl.oauthbearer.expected.audience",
+            "sasl.oauthbearer.expected.issuer",
+            "sasl.oauthbearer.jwks.endpoint.refresh.ms",
+            "sasl.oauthbearer.jwks.endpoint.retry.backoff.max.ms",
+            "sasl.oauthbearer.jwks.endpoint.retry.backoff.ms",
+            "sasl.oauthbearer.jwks.endpoint.url",
+            "sasl.oauthbearer.scope.claim.name",
+            "sasl.oauthbearer.sub.claim.name",
+            "sasl.oauthbearer.token.endpoint.url",
+            "sasl.server.callback.handler.class",
+            "sasl.server.max.receive.size",
+            "security.providers",
+            "server.max.startup.time.ms",
+            "socket.connection.setup.timeout.max.ms",
+            "socket.connection.setup.timeout.ms",
+            "socket.listen.backlog.size",
+            "socket.receive.buffer.bytes",
+            "socket.request.max.bytes",
+            "socket.send.buffer.bytes",
+            "ssl.cipher.suites",
+            "ssl.client.auth",
+            "ssl.enabled.protocols",
+            "ssl.endpoint.identification.algorithm",
+            "ssl.engine.factory.class",
+            "ssl.key.password",
+            "ssl.keymanager.algorithm",
+            "ssl.keystore.certificate.chain",
+            "ssl.keystore.key",
+            "ssl.keystore.location",
+            "ssl.keystore.password",
+            "ssl.keystore.type",
+            "ssl.principal.mapping.rules",
+            "ssl.protocol",
+            "ssl.provider",
+            "ssl.secure.random.implementation",
+            "ssl.trustmanager.algorithm",
+            "ssl.truststore.certificates",
+            "ssl.truststore.location",
+            "ssl.truststore.password",
+            "ssl.truststore.type",
+            "super.users",
+            "transaction.state.log.min.isr",
+            "transaction.state.log.replication.factor",
+            "queued.max.requests",
+            "queued.max.requests.bytes",
+            "unclean.leader.election.enable"
+    );
+
     /**
      * Constructor
      *
      * @param reconciliation    Reconciliation marker
-     * @param brokerConfigs     Broker configuration from Kafka Admin API
+     * @param nodeConfigs       Kafka node configuration from Kafka Admin API
      * @param desired           Desired configuration
      * @param kafkaVersion      Kafka version
-     * @param brokerNodeRef     Broker node reference
+     * @param nodeRef           Node reference
+     * @param isController      Whether node has a controller role
+     * @param isBroker          Whether node has a broker role
      */
-    protected KafkaBrokerConfigurationDiff(Reconciliation reconciliation, Config brokerConfigs, String desired, KafkaVersion kafkaVersion, NodeRef brokerNodeRef) {
+    protected KafkaConfigurationDiff(Reconciliation reconciliation, Config nodeConfigs, String desired, KafkaVersion kafkaVersion, NodeRef nodeRef, boolean isController, boolean isBroker) {
```

Comment:
the `NodeRef` instance expose `controller` and `broker` boolean, why do we have to pass two additional booleans for that?